rust/signed_doc/src/metadata/extra_fields.rs

```diff
@@ -160,12 +159,14 @@ impl ExtraFields {
     }
 
     /// Converting COSE Protected Header to `ExtraFields`.
+    /// Return `None` if it fails during
     #[allow(clippy::too_many_lines)]
     pub(crate) fn from_protected_header(
         protected: &ProtectedHeader, error_report: &ProblemReport,
-    ) -> anyhow::Result<Self> {
+    ) -> Option<Self> {
         /// Context for error messages.
         const CONTEXT: &str = "COSE ProtectedHeader to ExtraFields";
+        let mut valid = false;
```

Comment:
Is this an intended approach? valid is never set to true and why do we need this when we can check the `is_problematic()`

Reply:
Sorry, my mistake, by default it should be `true`, fixed it.

Reply:
>why do we need this when we can check the `is_problematic()`

We can't use `is_problematic()` because its possible the `report` which was passed as an argument was already problematic, and in this case even if inside this function everything is fine, we will return an error which is wrong